src/sagas.js

```diff
@@ -163,37 +175,62 @@ function* fetchIcon ({ iconId }) {
 }
 
 function* refreshData (action) {
-  console.log('refreshData(): ', action.payload)
-  if (!action.payload.quiet) {
-    logDebug('refreshData(): not quiet')
+  logDebug('refreshData(): ', action.payload)
+
+  // refresh VMs and remove any that haven't been refreshed
+  const vmsFetchedById = []
+  for (let vmId of Selectors.getVmsFetchedById().toArray()) {
```

Comment:
How about fetching by count first and then subtract fetched-by-count-vm-ids from `Selectors.getVmsFetchedById()` and fetch only the ids from subtraction result? It would prevent fetch duplicates.

Reply:
@mareklibra has a similar (or really the same) suggestion.  I'm gonna try it out and see how it looks.